{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Great, now that we discussed a little let's continue"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Given that the current approach utilized by the authors lacks reproducibility, we will explore an alternative method by leveraging nf-core pipelines for data analysis.\n",
    "\n",
    "Please explain, how we will achieve reproducibility for the course  with this approach.\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "By running the rnaseq pipeline from nf-core. "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "You have successfully downloaded 2 of the fastq files we will use in our study.\n",
    "\n",
    "What is the next step if we want to first have a count table and check the quality of our fastq files? What is the pipeline called to do so?"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "rnaseq"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Analyze the 2 files using an nf-core pipeline.\n",
    "\n",
    "What does this pipeline do?\n",
    "\n",
    "Which are the main tools that will be used in the pipeline?"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "rnaseq steps:\n",
    "\n",
    "1. Preprocessing \n",
    "        Merge re-sequenced FastQ files (cat)\n",
    "        Auto-infer strandedness by subsampling and pseudoalignment (fq, Salmon)\n",
    "        Read QC (FastQC)\n",
    "        UMI extraction (UMI-tools)\n",
    "        Adapter and quality trimming (Trim Galore!)\n",
    "        Removal of genome contaminants (BBSplit)\n",
    "        Removal of ribosomal RNA (SortMeRNA)\n",
    "2. Genome alignment & quantification\n",
    "        Choice of multiple alignment and quantification routes (For STAR the sentieon implementation can be chosen):\n",
    "        STAR -> Salmon\n",
    "        STAR -> RSEM\n",
    "        HiSAT2 -> NO QUANTIFICATION -> used in the paper\n",
    "3. Pseudo-alignment \n",
    "    Salmon or Kallisto\n",
    "4. Post-processing\n",
    "    Sort and index alignments (SAMtools)\n",
    "    UMI-based deduplication (UMI-tools)\n",
    "    Duplicate read marking (picard MarkDuplicates)\n",
    "    Transcript assembly and quantification (StringTie)\n",
    "    Create bigWig coverage files (BEDTools, bedGraphToBigWig)\n",
    "5. Final QC\n",
    "    RSeQC\n",
    "    Qualimap\n",
    "    dupRadar\n",
    "    Preseq\n",
    "    DESeq2\n",
    "    Kraken2 \n",
    "    MultiQC\n",
    "\n",
    "\n",
    "    The paper only reports using HISAT2, HT-Seq33 and the DESeq2 R package. "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "As all other nf-core pipelines, the chosen pipeline takes in a samplesheet as input.\n",
    "\n",
    "Use Python and pandas to create the samplesheet for your 2 samples. Feel free to make use of the table you created earlier today.\n",
    "\n",
    "Choose your sample names wisely, they must be the connection of the results to the metadata. If you can't find the sample in the metadata later, the analysis was useless."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "There was a samplesheet provided by fetchngs - only a few syntax corrections had to be made"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[33mNextflow 25.04.8 is available - Please consider updating your version to it\u001b[m\n",
      "\n",
      "\u001b[1m\u001b[38;5;232m\u001b[48;5;43m N E X T F L O W \u001b[0;2m  ~  \u001b[mversion 25.04.7\u001b[m\n",
      "\u001b[K\n",
      "Launching\u001b[35m `https://github.com/nf-core/rnaseq` \u001b[0;2m[\u001b[0;1;36mgigantic_dalembert\u001b[0;2m] DSL2 - \u001b[36mrevision: \u001b[0;36m9738a2df42 [master]\u001b[m\n",
      "\u001b[K\n",
      "\n",
      "------------------------------------------------------\n",
      "                                        ,--./,-.\n",
      "        ___     __   __   __   ___     /,-._.--~'\n",
      "  |\\ | |__  __ /  ` /  \\ |__) |__         }  {\n",
      "  | \\| |       \\__, \\__/ |  \\ |___     \\`-._,-`-,\n",
      "                                        `._,._,'\n",
      "  nf-core/rnaseq 3.21.0\n",
      "------------------------------------------------------\n",
      "\u001b[1mInput/output options\u001b[0m\n",
      "  \u001b[0;34minput              : \u001b[0;32m./SRFetch_results/samplesheet/samplesheet_fixed.csv\u001b[0m\n",
      "  \u001b[0;34moutdir             : \u001b[0;32m./rnaseq_results\u001b[0m\n",
      "\n",
      "\u001b[1mReference genome options\u001b[0m\n",
      "  \u001b[0;34mgenome             : \u001b[0;32mGRCm38\u001b[0m\n",
      "  \u001b[0;34mfasta              : \u001b[0;32ms3://ngi-igenomes/igenomes//Mus_musculus/Ensembl/GRCm38/Sequence/WholeGenomeFasta/genome.fa\u001b[0m\n",
      "  \u001b[0;34mgtf                : \u001b[0;32ms3://ngi-igenomes/igenomes//Mus_musculus/Ensembl/GRCm38/Annotation/Genes/genes.gtf\u001b[0m\n",
      "  \u001b[0;34mgene_bed           : \u001b[0;32ms3://ngi-igenomes/igenomes//Mus_musculus/Ensembl/GRCm38/Annotation/Genes/genes.bed\u001b[0m\n",
      "  \u001b[0;34mstar_index         : \u001b[0;32ms3://ngi-igenomes/igenomes//Mus_musculus/Ensembl/GRCm38/Sequence/STARIndex/\u001b[0m\n",
      "\n",
      "\u001b[1mUMI options\u001b[0m\n",
      "  \u001b[0;34mumi_discard_read   : \u001b[0;32m0\u001b[0m\n",
      "\n",
      "\u001b[1mAlignment options\u001b[0m\n",
      "  \u001b[0;34maligner            : \u001b[0;32mhisat2\u001b[0m\n",
      "\n",
      "\u001b[1mGeneric options\u001b[0m\n",
      "  \u001b[0;34mtrace_report_suffix: \u001b[0;32m2025-10-16_20-10-08\u001b[0m\n",
      "\n",
      "\u001b[1mCore Nextflow options\u001b[0m\n",
      "  \u001b[0;34mrevision           : \u001b[0;32mmaster\u001b[0m\n",
      "  \u001b[0;34mrunName            : \u001b[0;32mgigantic_dalembert\u001b[0m\n",
      "  \u001b[0;34mcontainerEngine    : \u001b[0;32mdocker\u001b[0m\n",
      "  \u001b[0;34mlaunchDir          : \u001b[0;32m/home/clara/computational-workflows-2025/notebooks/day_02\u001b[0m\n",
      "  \u001b[0;34mworkDir            : \u001b[0;32m/home/clara/computational-workflows-2025/notebooks/day_02/work\u001b[0m\n",
      "  \u001b[0;34mprojectDir         : \u001b[0;32m/home/clara/.nextflow/assets/nf-core/rnaseq\u001b[0m\n",
      "  \u001b[0;34muserName           : \u001b[0;32mclara\u001b[0m\n",
      "  \u001b[0;34mprofile            : \u001b[0;32mdocker\u001b[0m\n",
      "  \u001b[0;34mconfigFiles        : \u001b[0;32m/home/clara/.nextflow/assets/nf-core/rnaseq/nextflow.config, /home/clara/computational-workflows-2025/notebooks/day_02/nextflow.config\u001b[0m\n",
      "\n",
      "!! Only displaying parameters that differ from the pipeline defaults !!\n",
      "-\u001b[2m----------------------------------------------------\u001b[0m-\n",
      "* The pipeline\n",
      "    https://doi.org/10.5281/zenodo.1400710\n",
      "\n",
      "* The nf-core framework\n",
      "    https://doi.org/10.1038/s41587-020-0439-x\n",
      "\n",
      "* Software dependencies\n",
      "    https://github.com/nf-core/rnaseq/blob/master/CITATIONS.md\n",
      "\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…:PREPARE_GENOME:\u001b[mGTF_FILTER -\u001b[K\n",
      "\u001b[33mWARN: The operator `first` is useless when applied to a value channel which returns a single value by definition\u001b[39m\u001b[K\n",
      "\u001b[3A\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…:PREPARE_GENOME:\u001b[mGTF_FILTER -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…OME:\u001b[mMAKE_TRANSCRIPTS_FASTA -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…ENOME:\u001b[mCUSTOM_GETCHROMSIZES -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…:\u001b[mHISAT2_EXTRACTSPLICESITES -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…REPARE_GENOME:\u001b[mHISAT2_BUILD -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…ASEQ:RNASEQ:\u001b[mSAMTOOLS_INDEX -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…_SETSTRANDEDNESS:\u001b[mCAT_FASTQ -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…ER_SETSTRANDEDNESS:\u001b[mFQ_LINT -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…UMITOOLS_TRIMGALORE:\u001b[mFASTQC -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…OOLS_TRIMGALORE:\u001b[mTRIMGALORE -\u001b[K\n",
      "\u001b[33mWARN: The operator `first` is useless when applied to a value channel which returns a single value by definition\u001b[39m\u001b[K\n",
      "\u001b[33mWARN: Found the following unidentified headers in /home/clara/computational-workflows-2025/notebooks/day_02/SRFetch_results/samplesheet/samplesheet_fixed.csv:\u001b[K\n",
      "\t- run_accession\u001b[K\n",
      "\t- experiment_accession\u001b[K\n",
      "\t- sample_accession\u001b[K\n",
      "\t- secondary_sample_accession\u001b[K\n",
      "\t- study_accession\u001b[K\n",
      "\t- secondary_study_accession\u001b[K\n",
      "\t- submission_accession\u001b[K\n",
      "\t- run_alias\u001b[K\n",
      "\t- experiment_alias\u001b[K\n",
      "\t- sample_alias\u001b[K\n",
      "\t- study_alias\u001b[K\n",
      "\t- library_layout\u001b[K\n",
      "\t- library_selection\u001b[K\n",
      "\t- library_source\u001b[K\n",
      "\t- library_strategy\u001b[K\n",
      "\t- library_name\u001b[K\n",
      "\t- instrument_model\u001b[K\n",
      "\t- instrument_platform\u001b[K\n",
      "\t- base_count\u001b[K\n",
      "\t- read_count\u001b[K\n",
      "\t- tax_id\u001b[K\n",
      "\t- scientific_name\u001b[K\n",
      "\t- sample_title\u001b[K\n",
      "\t- experiment_title\u001b[K\n",
      "\t- study_title\u001b[K\n",
      "\t- sample_description\u001b[K\n",
      "\t- fastq_md5\u001b[K\n",
      "\t- fastq_bytes\u001b[K\n",
      "\t- fastq_ftp\u001b[K\n",
      "\t- fastq_galaxy\u001b[K\n",
      "\t- fastq_aspera\u001b[K\n",
      "\u001b[39m\u001b[K\n",
      "\u001b[45A\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…:PREPARE_GENOME:\u001b[mGTF_FILTER -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…OME:\u001b[mMAKE_TRANSCRIPTS_FASTA -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…ENOME:\u001b[mCUSTOM_GETCHROMSIZES -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…:\u001b[mHISAT2_EXTRACTSPLICESITES -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…REPARE_GENOME:\u001b[mHISAT2_BUILD -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…ASEQ:RNASEQ:\u001b[mSAMTOOLS_INDEX -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…_SETSTRANDEDNESS:\u001b[mCAT_FASTQ -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…ER_SETSTRANDEDNESS:\u001b[mFQ_LINT -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…UMITOOLS_TRIMGALORE:\u001b[mFASTQC -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…OOLS_TRIMGALORE:\u001b[mTRIMGALORE -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…ESS:\u001b[mFQ_LINT_AFTER_TRIMMING -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…PLE_FQ_SALMON:\u001b[mSALMON_INDEX -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…PLE_FQ_SALMON:\u001b[mFQ_SUBSAMPLE -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…PLE_FQ_SALMON:\u001b[mSALMON_QUANT -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…_ALIGN_HISAT2:\u001b[mHISAT2_ALIGN -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…ATS_SAMTOOLS:\u001b[mSAMTOOLS_SORT -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…TS_SAMTOOLS:\u001b[mSAMTOOLS_INDEX -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…TS_SAMTOOLS:\u001b[mSAMTOOLS_STATS -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…SAMTOOLS:\u001b[mSAMTOOLS_FLAGSTAT -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…SAMTOOLS:\u001b[mSAMTOOLS_IDXSTATS -\u001b[K\n",
      "\u001b[3;2mPlus \u001b[1m5\u001b[0;3;2m more processes waiting for tasks…\u001b[m\u001b[K\n",
      "\u001b[33mWARN: The operator `first` is useless when applied to a value channel which returns a single value by definition\u001b[39m\u001b[K\n",
      "\u001b[33mWARN: Found the following unidentified headers in /home/clara/computational-workflows-2025/notebooks/day_02/SRFetch_results/samplesheet/samplesheet_fixed.csv:\u001b[K\n",
      "\t- run_accession\u001b[K\n",
      "\t- experiment_accession\u001b[K\n",
      "\t- sample_accession\u001b[K\n",
      "\t- secondary_sample_accession\u001b[K\n",
      "\t- study_accession\u001b[K\n",
      "\t- secondary_study_accession\u001b[K\n",
      "\t- submission_accession\u001b[K\n",
      "\t- run_alias\u001b[K\n",
      "\t- experiment_alias\u001b[K\n",
      "\t- sample_alias\u001b[K\n",
      "\t- study_alias\u001b[K\n",
      "\t- library_layout\u001b[K\n",
      "\t- library_selection\u001b[K\n",
      "\t- library_source\u001b[K\n",
      "\t- library_strategy\u001b[K\n",
      "\t- library_name\u001b[K\n",
      "\t- instrument_model\u001b[K\n",
      "\t- instrument_platform\u001b[K\n",
      "\t- base_count\u001b[K\n",
      "\t- read_count\u001b[K\n",
      "\t- tax_id\u001b[K\n",
      "\t- scientific_name\u001b[K\n",
      "\t- sample_title\u001b[K\n",
      "\t- experiment_title\u001b[K\n",
      "\t- study_title\u001b[K\n",
      "\t- sample_description\u001b[K\n",
      "\t- fastq_md5\u001b[K\n",
      "\t- fastq_bytes\u001b[K\n",
      "\t- fastq_ftp\u001b[K\n",
      "\t- fastq_galaxy\u001b[K\n",
      "\t- fastq_aspera\u001b[K\n",
      "\u001b[39m\u001b[K\n",
      "\u001b[56A\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…:PREPARE_GENOME:\u001b[mGTF_FILTER -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…OME:\u001b[mMAKE_TRANSCRIPTS_FASTA -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…ENOME:\u001b[mCUSTOM_GETCHROMSIZES -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…:\u001b[mHISAT2_EXTRACTSPLICESITES -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…REPARE_GENOME:\u001b[mHISAT2_BUILD -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…ASEQ:RNASEQ:\u001b[mSAMTOOLS_INDEX -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…_SETSTRANDEDNESS:\u001b[mCAT_FASTQ -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…ER_SETSTRANDEDNESS:\u001b[mFQ_LINT -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…UMITOOLS_TRIMGALORE:\u001b[mFASTQC -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…OOLS_TRIMGALORE:\u001b[mTRIMGALORE -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…ESS:\u001b[mFQ_LINT_AFTER_TRIMMING -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…PLE_FQ_SALMON:\u001b[mSALMON_INDEX -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…PLE_FQ_SALMON:\u001b[mFQ_SUBSAMPLE -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…PLE_FQ_SALMON:\u001b[mSALMON_QUANT -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…_ALIGN_HISAT2:\u001b[mHISAT2_ALIGN -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…ATS_SAMTOOLS:\u001b[mSAMTOOLS_SORT -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…TS_SAMTOOLS:\u001b[mSAMTOOLS_INDEX -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…TS_SAMTOOLS:\u001b[mSAMTOOLS_STATS -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…SAMTOOLS:\u001b[mSAMTOOLS_FLAGSTAT -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…SAMTOOLS:\u001b[mSAMTOOLS_IDXSTATS -\u001b[K\n",
      "\u001b[3;2mPlus \u001b[1m21\u001b[0;3;2m more processes waiting for tasks…\u001b[m\u001b[K\n",
      "\u001b[33mWARN: The operator `first` is useless when applied to a value channel which returns a single value by definition\u001b[39m\u001b[K\n",
      "\u001b[33mWARN: Found the following unidentified headers in /home/clara/computational-workflows-2025/notebooks/day_02/SRFetch_results/samplesheet/samplesheet_fixed.csv:\u001b[K\n",
      "\t- run_accession\u001b[K\n",
      "\t- experiment_accession\u001b[K\n",
      "\t- sample_accession\u001b[K\n",
      "\t- secondary_sample_accession\u001b[K\n",
      "\t- study_accession\u001b[K\n",
      "\t- secondary_study_accession\u001b[K\n",
      "\t- submission_accession\u001b[K\n",
      "\t- run_alias\u001b[K\n",
      "\t- experiment_alias\u001b[K\n",
      "\t- sample_alias\u001b[K\n",
      "\t- study_alias\u001b[K\n",
      "\t- library_layout\u001b[K\n",
      "\t- library_selection\u001b[K\n",
      "\t- library_source\u001b[K\n",
      "\t- library_strategy\u001b[K\n",
      "\t- library_name\u001b[K\n",
      "\t- instrument_model\u001b[K\n",
      "\t- instrument_platform\u001b[K\n",
      "\t- base_count\u001b[K\n",
      "\t- read_count\u001b[K\n",
      "\t- tax_id\u001b[K\n",
      "\t- scientific_name\u001b[K\n",
      "\t- sample_title\u001b[K\n",
      "\t- experiment_title\u001b[K\n",
      "\t- study_title\u001b[K\n",
      "\t- sample_description\u001b[K\n",
      "\t- fastq_md5\u001b[K\n",
      "\t- fastq_bytes\u001b[K\n",
      "\t- fastq_ftp\u001b[K\n",
      "\t- fastq_galaxy\u001b[K\n",
      "\t- fastq_aspera\u001b[K\n",
      "\u001b[39m\u001b[K\n",
      "\u001b[56A\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…:PREPARE_GENOME:\u001b[mGTF_FILTER -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…OME:\u001b[mMAKE_TRANSCRIPTS_FASTA -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…ENOME:\u001b[mCUSTOM_GETCHROMSIZES -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…:\u001b[mHISAT2_EXTRACTSPLICESITES -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…REPARE_GENOME:\u001b[mHISAT2_BUILD -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…ASEQ:RNASEQ:\u001b[mSAMTOOLS_INDEX -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…_SETSTRANDEDNESS:\u001b[mCAT_FASTQ -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…ER_SETSTRANDEDNESS:\u001b[mFQ_LINT -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…UMITOOLS_TRIMGALORE:\u001b[mFASTQC -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…OOLS_TRIMGALORE:\u001b[mTRIMGALORE -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…ESS:\u001b[mFQ_LINT_AFTER_TRIMMING -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…PLE_FQ_SALMON:\u001b[mSALMON_INDEX -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…PLE_FQ_SALMON:\u001b[mFQ_SUBSAMPLE -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…PLE_FQ_SALMON:\u001b[mSALMON_QUANT -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…_ALIGN_HISAT2:\u001b[mHISAT2_ALIGN -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…ATS_SAMTOOLS:\u001b[mSAMTOOLS_SORT -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…TS_SAMTOOLS:\u001b[mSAMTOOLS_INDEX -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…TS_SAMTOOLS:\u001b[mSAMTOOLS_STATS -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…SAMTOOLS:\u001b[mSAMTOOLS_FLAGSTAT -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…SAMTOOLS:\u001b[mSAMTOOLS_IDXSTATS -\u001b[K\n",
      "\u001b[3;2mPlus \u001b[1m24\u001b[0;3;2m more processes waiting for tasks…\u001b[m\u001b[K\n",
      "\u001b[33mWARN: The operator `first` is useless when applied to a value channel which returns a single value by definition\u001b[39m\u001b[K\n",
      "\u001b[33mWARN: Found the following unidentified headers in /home/clara/computational-workflows-2025/notebooks/day_02/SRFetch_results/samplesheet/samplesheet_fixed.csv:\u001b[K\n",
      "\t- run_accession\u001b[K\n",
      "\t- experiment_accession\u001b[K\n",
      "\t- sample_accession\u001b[K\n",
      "\t- secondary_sample_accession\u001b[K\n",
      "\t- study_accession\u001b[K\n",
      "\t- secondary_study_accession\u001b[K\n",
      "\t- submission_accession\u001b[K\n",
      "\t- run_alias\u001b[K\n",
      "\t- experiment_alias\u001b[K\n",
      "\t- sample_alias\u001b[K\n",
      "\t- study_alias\u001b[K\n",
      "\t- library_layout\u001b[K\n",
      "\t- library_selection\u001b[K\n",
      "\t- library_source\u001b[K\n",
      "\t- library_strategy\u001b[K\n",
      "\t- library_name\u001b[K\n",
      "\t- instrument_model\u001b[K\n",
      "\t- instrument_platform\u001b[K\n",
      "\t- base_count\u001b[K\n",
      "\t- read_count\u001b[K\n",
      "\t- tax_id\u001b[K\n",
      "\t- scientific_name\u001b[K\n",
      "\t- sample_title\u001b[K\n",
      "\t- experiment_title\u001b[K\n",
      "\t- study_title\u001b[K\n",
      "\t- sample_description\u001b[K\n",
      "\t- fastq_md5\u001b[K\n",
      "\t- fastq_bytes\u001b[K\n",
      "\t- fastq_ftp\u001b[K\n",
      "\t- fastq_galaxy\u001b[K\n",
      "\t- fastq_aspera\u001b[K\n",
      "\u001b[39m\u001b[K\n",
      "\u001b[56A\n",
      "\u001b[2mexecutor >  local (2)\u001b[m\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…:PREPARE_GENOME:\u001b[mGTF_FILTER -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…OME:\u001b[mMAKE_TRANSCRIPTS_FASTA -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…ENOME:\u001b[mCUSTOM_GETCHROMSIZES -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…:\u001b[mHISAT2_EXTRACTSPLICESITES -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…REPARE_GENOME:\u001b[mHISAT2_BUILD -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…ASEQ:RNASEQ:\u001b[mSAMTOOLS_INDEX -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…_SETSTRANDEDNESS:\u001b[mCAT_FASTQ -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m50/661922\u001b[0;2m] \u001b[0;2mNFC…NESS:\u001b[mFQ_LINT\u001b[33;2m (\u001b[0;33mSRX19144488\u001b[2m)\u001b[m\u001b[2m |\u001b[m 0 of 2\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…UMITOOLS_TRIMGALORE:\u001b[mFASTQC -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…OOLS_TRIMGALORE:\u001b[mTRIMGALORE -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…ESS:\u001b[mFQ_LINT_AFTER_TRIMMING -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…PLE_FQ_SALMON:\u001b[mSALMON_INDEX -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…PLE_FQ_SALMON:\u001b[mFQ_SUBSAMPLE -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…PLE_FQ_SALMON:\u001b[mSALMON_QUANT -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…_ALIGN_HISAT2:\u001b[mHISAT2_ALIGN -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…ATS_SAMTOOLS:\u001b[mSAMTOOLS_SORT -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…TS_SAMTOOLS:\u001b[mSAMTOOLS_INDEX -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…TS_SAMTOOLS:\u001b[mSAMTOOLS_STATS -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…SAMTOOLS:\u001b[mSAMTOOLS_FLAGSTAT -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…SAMTOOLS:\u001b[mSAMTOOLS_IDXSTATS -\u001b[K\n",
      "\u001b[3;2mPlus \u001b[1m24\u001b[0;3;2m more processes waiting for tasks…\u001b[m\u001b[K\n",
      "\u001b[33mWARN: The operator `first` is useless when applied to a value channel which returns a single value by definition\u001b[39m\u001b[K\n",
      "\u001b[33mWARN: Found the following unidentified headers in /home/clara/computational-workflows-2025/notebooks/day_02/SRFetch_results/samplesheet/samplesheet_fixed.csv:\u001b[K\n",
      "\t- run_accession\u001b[K\n",
      "\t- experiment_accession\u001b[K\n",
      "\t- sample_accession\u001b[K\n",
      "\t- secondary_sample_accession\u001b[K\n",
      "\t- study_accession\u001b[K\n",
      "\t- secondary_study_accession\u001b[K\n",
      "\t- submission_accession\u001b[K\n",
      "\t- run_alias\u001b[K\n",
      "\t- experiment_alias\u001b[K\n",
      "\t- sample_alias\u001b[K\n",
      "\t- study_alias\u001b[K\n",
      "\t- library_layout\u001b[K\n",
      "\t- library_selection\u001b[K\n",
      "\t- library_source\u001b[K\n",
      "\t- library_strategy\u001b[K\n",
      "\t- library_name\u001b[K\n",
      "\t- instrument_model\u001b[K\n",
      "\t- instrument_platform\u001b[K\n",
      "\t- base_count\u001b[K\n",
      "\t- read_count\u001b[K\n",
      "\t- tax_id\u001b[K\n",
      "\t- scientific_name\u001b[K\n",
      "\t- sample_title\u001b[K\n",
      "\t- experiment_title\u001b[K\n",
      "\t- study_title\u001b[K\n",
      "\t- sample_description\u001b[K\n",
      "\t- fastq_md5\u001b[K\n",
      "\t- fastq_bytes\u001b[K\n",
      "\t- fastq_ftp\u001b[K\n",
      "\t- fastq_galaxy\u001b[K\n",
      "\t- fastq_aspera\u001b[K\n",
      "\u001b[39m\u001b[K\n",
      "\u001b[57A\n",
      "\u001b[2mexecutor >  local (2)\u001b[m\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…:PREPARE_GENOME:\u001b[mGTF_FILTER -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…OME:\u001b[mMAKE_TRANSCRIPTS_FASTA -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…ENOME:\u001b[mCUSTOM_GETCHROMSIZES -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…:\u001b[mHISAT2_EXTRACTSPLICESITES -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…REPARE_GENOME:\u001b[mHISAT2_BUILD -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…ASEQ:RNASEQ:\u001b[mSAMTOOLS_INDEX -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…_SETSTRANDEDNESS:\u001b[mCAT_FASTQ -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m50/661922\u001b[0;2m] \u001b[0;2mNFC…NESS:\u001b[mFQ_LINT\u001b[33;2m (\u001b[0;33mSRX19144488\u001b[2m)\u001b[m\u001b[2m |\u001b[m 0 of 2\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…UMITOOLS_TRIMGALORE:\u001b[mFASTQC -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…OOLS_TRIMGALORE:\u001b[mTRIMGALORE -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…ESS:\u001b[mFQ_LINT_AFTER_TRIMMING -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…PLE_FQ_SALMON:\u001b[mSALMON_INDEX -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…PLE_FQ_SALMON:\u001b[mFQ_SUBSAMPLE -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…PLE_FQ_SALMON:\u001b[mSALMON_QUANT -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…_ALIGN_HISAT2:\u001b[mHISAT2_ALIGN -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…ATS_SAMTOOLS:\u001b[mSAMTOOLS_SORT -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…TS_SAMTOOLS:\u001b[mSAMTOOLS_INDEX -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…TS_SAMTOOLS:\u001b[mSAMTOOLS_STATS -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…SAMTOOLS:\u001b[mSAMTOOLS_FLAGSTAT -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…SAMTOOLS:\u001b[mSAMTOOLS_IDXSTATS -\u001b[K\n",
      "\u001b[3;2mPlus \u001b[1m24\u001b[0;3;2m more processes waiting for tasks…\u001b[m\u001b[K\n",
      "Staging foreign file: s3://ngi-igenomes/igenomes/Mus_musculus/Ensembl/GRCm38/Sequence/WholeGenomeFasta/genome.fa\u001b[K\n",
      "Staging foreign file: s3://ngi-igenomes/igenomes/Mus_musculus/Ensembl/GRCm38/Annotation/Genes/genes.gtf\u001b[K\n",
      "\u001b[33mWARN: The operator `first` is useless when applied to a value channel which returns a single value by definition\u001b[39m\u001b[K\n",
      "\u001b[33mWARN: Found the following unidentified headers in /home/clara/computational-workflows-2025/notebooks/day_02/SRFetch_results/samplesheet/samplesheet_fixed.csv:\u001b[K\n",
      "\t- run_accession\u001b[K\n",
      "\t- experiment_accession\u001b[K\n",
      "\t- sample_accession\u001b[K\n",
      "\t- secondary_sample_accession\u001b[K\n",
      "\t- study_accession\u001b[K\n",
      "\t- secondary_study_accession\u001b[K\n",
      "\t- submission_accession\u001b[K\n",
      "\t- run_alias\u001b[K\n",
      "\t- experiment_alias\u001b[K\n",
      "\t- sample_alias\u001b[K\n",
      "\t- study_alias\u001b[K\n",
      "\t- library_layout\u001b[K\n",
      "\t- library_selection\u001b[K\n",
      "\t- library_source\u001b[K\n",
      "\t- library_strategy\u001b[K\n",
      "\t- library_name\u001b[K\n",
      "\t- instrument_model\u001b[K\n",
      "\t- instrument_platform\u001b[K\n",
      "\t- base_count\u001b[K\n",
      "\t- read_count\u001b[K\n",
      "\t- tax_id\u001b[K\n",
      "\t- scientific_name\u001b[K\n",
      "\t- sample_title\u001b[K\n",
      "\t- experiment_title\u001b[K\n",
      "\t- study_title\u001b[K\n",
      "\t- sample_description\u001b[K\n",
      "\t- fastq_md5\u001b[K\n",
      "\t- fastq_bytes\u001b[K\n",
      "\t- fastq_ftp\u001b[K\n",
      "\t- fastq_galaxy\u001b[K\n",
      "\t- fastq_aspera\u001b[K\n",
      "\u001b[39m\u001b[K\n",
      "\u001b[59A\n",
      "\u001b[2mexecutor >  local (2)\u001b[m\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…:PREPARE_GENOME:\u001b[mGTF_FILTER -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…OME:\u001b[mMAKE_TRANSCRIPTS_FASTA -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…ENOME:\u001b[mCUSTOM_GETCHROMSIZES -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…:\u001b[mHISAT2_EXTRACTSPLICESITES -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…REPARE_GENOME:\u001b[mHISAT2_BUILD -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…ASEQ:RNASEQ:\u001b[mSAMTOOLS_INDEX -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…_SETSTRANDEDNESS:\u001b[mCAT_FASTQ -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m32/1484b4\u001b[0;2m] \u001b[0;2mNFC…NESS:\u001b[mFQ_LINT\u001b[33;2m (\u001b[0;33mSRX19144486\u001b[2m)\u001b[m\u001b[2m |\u001b[m 1 of 2\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…UMITOOLS_TRIMGALORE:\u001b[mFASTQC\u001b[2m |\u001b[m 0 of 1\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…OOLS_TRIMGALORE:\u001b[mTRIMGALORE\u001b[2m |\u001b[m 0 of 1\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…ESS:\u001b[mFQ_LINT_AFTER_TRIMMING -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…PLE_FQ_SALMON:\u001b[mSALMON_INDEX -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…PLE_FQ_SALMON:\u001b[mFQ_SUBSAMPLE -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…PLE_FQ_SALMON:\u001b[mSALMON_QUANT -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…_ALIGN_HISAT2:\u001b[mHISAT2_ALIGN -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…ATS_SAMTOOLS:\u001b[mSAMTOOLS_SORT -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…TS_SAMTOOLS:\u001b[mSAMTOOLS_INDEX -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…TS_SAMTOOLS:\u001b[mSAMTOOLS_STATS -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…SAMTOOLS:\u001b[mSAMTOOLS_FLAGSTAT -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…SAMTOOLS:\u001b[mSAMTOOLS_IDXSTATS -\u001b[K\n",
      "\u001b[3;2mPlus \u001b[1m24\u001b[0;3;2m more processes waiting for tasks…\u001b[m\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[59A\n",
      "\u001b[2mexecutor >  local (3)\u001b[m\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…:PREPARE_GENOME:\u001b[mGTF_FILTER -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…OME:\u001b[mMAKE_TRANSCRIPTS_FASTA -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…ENOME:\u001b[mCUSTOM_GETCHROMSIZES -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…:\u001b[mHISAT2_EXTRACTSPLICESITES -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…REPARE_GENOME:\u001b[mHISAT2_BUILD -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…ASEQ:RNASEQ:\u001b[mSAMTOOLS_INDEX -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…_SETSTRANDEDNESS:\u001b[mCAT_FASTQ -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m50/661922\u001b[0;2m] \u001b[0;2mNFC…NESS:\u001b[mFQ_LINT\u001b[33;2m (\u001b[0;33mSRX19144488\u001b[2m)\u001b[m\u001b[2m |\u001b[m 2 of 2\u001b[32m ✔\u001b[m\u001b[K\n",
      "\u001b[2m[\u001b[0;34m58/f04eff\u001b[0;2m] \u001b[0;2mNFC…ALORE:\u001b[mFASTQC\u001b[33;2m (\u001b[0;33mSRX19144486\u001b[2m)\u001b[m\u001b[2m |\u001b[m 0 of 2\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…OOLS_TRIMGALORE:\u001b[mTRIMGALORE\u001b[2m |\u001b[m 0 of 2\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…ESS:\u001b[mFQ_LINT_AFTER_TRIMMING -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…PLE_FQ_SALMON:\u001b[mSALMON_INDEX -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…PLE_FQ_SALMON:\u001b[mFQ_SUBSAMPLE -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…PLE_FQ_SALMON:\u001b[mSALMON_QUANT -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…_ALIGN_HISAT2:\u001b[mHISAT2_ALIGN -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…ATS_SAMTOOLS:\u001b[mSAMTOOLS_SORT -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…TS_SAMTOOLS:\u001b[mSAMTOOLS_INDEX -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…TS_SAMTOOLS:\u001b[mSAMTOOLS_STATS -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…SAMTOOLS:\u001b[mSAMTOOLS_FLAGSTAT -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…SAMTOOLS:\u001b[mSAMTOOLS_IDXSTATS -\u001b[K\n",
      "\u001b[3;2mPlus \u001b[1m24\u001b[0;3;2m more processes waiting for tasks…\u001b[m\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n"
     ]
    }
   ],
   "source": [
    "# post here the command you used to run nf-core/rnaseq\n",
    "\n",
    "!nextflow run nf-core/rnaseq --input ./SRFetch_results/samplesheet/samplesheet_fixed.csv --outdir ./rnaseq_results --aligner hisat2 --genome GRCm38 -profile docker"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Note: This pipeline doesn't run for me. It takes forever and eventually, VS code freezes. I analyzed the results you provided for us later\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Explain all the parameters you set and why you set them in this way.\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "--input: A sample sheet we received from the previous run\n",
    "--aligner: rnaseq offers multiple choices, we used the one specified in the paper (hisat2)\n",
    "--genome: GRCm38 (mouse genome) used for reference"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Browsing the results"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "How did the pipeline perform?"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Overall, the pipeline performed pretty well. For most samples, the alignment coverage was quite good, with over 80% of reads mapped. The only exception is Nac_oxy_1, which had only 28% mapped reads. The error rate is also very low, in many cases below 1%. However, many samples contain a large amount of duplicates.  "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Explain the quality control steps. Are you happy with the quality and why. If not, why not.\n",
    "What are the possible steps that could lead to poorer results?\n",
    "Please give additional information on : \n",
    "- ribosomal rRNA: Is usually depleted, as it is not inetersting for this analysis. For most samples, rRNA% was accordingly very low (<1%). Exceptions are Nac_Sal_3 and Nac_Sal_7. A high rRNA percentage indicates insufficient depletion and potentially poorer transcriptome coverage. Unsurprisingly, these samples have a high duplication rate and low rate of proper pairs mapped. \n",
    "- Duplication: May happen due to over-amplification and may impact quantification accuracy. Duplication rates are very high for some samples.\n",
    "- GC content: A species-specific characteristic. For mammals, it should be somewhere between 40-45%. Largely skewed values could indicate biases in library preparation or contamination. For most samples, the GC content is around 50%, with the exception of Nac_oxy_1, which has a GC content of 63%.\n",
    "\n",
    "Overall, I think the pipeline performed alright from a technical standpoint. Most samples look fine in terms of reads mapped, error rate, rRNA%, and GC content. Duplication is a little high for most samples, potentially hinting at issues with RNA amplification.  \n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Would you exclude any samples? If yes, which and why?"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "I would exclude Nac_oxy_1, because of the low percentage of mapped reads and skewed GC content. Additionally, I would consider excluding Nac_Sal_3 and Nac_Sal_7 because of the high percentage of rRNA. "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "What would you now do to continue the experiment? What are the scientists trying to figure out? Which packages on R or python would you use?"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "I would continue with differential expression analysis to check for differentially expressed genes between conditions."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "workflows",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.13.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
